src/main/java/org/mockito/internal/MockitoCore.java

```diff
@@ -60,11 +69,36 @@ public boolean isTypeMockable(Class<?> typeToMock) {
         }
         MockSettingsImpl impl = MockSettingsImpl.class.cast(settings);
         MockCreationSettings<T> creationSettings = impl.build(typeToMock);
+        checkDoNotMockAnnotation(creationSettings.getTypeToMock(), creationSettings);
         T mock = createMock(creationSettings);
         mockingProgress().mockingStarted(mock, creationSettings);
         return mock;
     }
 
+    private void checkDoNotMockAnnotation(Class<?> typeToMock, MockCreationSettings<?> creationSettings) {
+        checkDoNotMockAnnotationForType(typeToMock);
+        creationSettings.getExtraInterfaces().forEach(MockitoCore::checkDoNotMockAnnotationForType);
+    }
+
+    private static void checkDoNotMockAnnotationForType(Class<?> type) {
+        if (type == null) {
```

Comment:
I suggest to drop this conditional, type cannot be null (unless I'm wrong :-)

Reply:
This will be `null` for `.getSuperClass()` of interfaces and `Object.class`. The tests will fail if we remove it.

Reply:
Maybe leave this as a comment

Reply:
Done